src/components/documentation/edit-tabs/edit-page-configuration.components.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { IScope } from 'angular';
+import '@gravitee/ui-components/wc/gv-switch';
+import { PageType } from '../../../services/documentation.service';
+
+interface IPageScope extends IScope {
+  fetcherJsonSchema: string;
+  rename: boolean;
+  editorReadonly: boolean;
+  currentTab: string;
+  currentTranslation: any;
+  acls: any;
+}
+
+const EditPageAclsComponent: ng.IComponentOptions = {
+  bindings: {
+    page: '<',
+  },
+  template: require('./edit-page-configuration.html'),
+  controller: function (Constants: any) {
+    'ngInject';
+
+    this.isSwagger = (): boolean => PageType.SWAGGER === this.page.type;
+    this.isMarkdownTemplate = (): boolean => PageType.MARKDOWN_TEMPLATE === this.page.type;
+
+    this.shouldShowOpenApiDocFormat = false;
+
+    this.$onInit = () => {
+      this.settings = Constants.env.settings;
+
+      this.shouldShowOpenApiDocFormat =
+        this.settings &&
+        this.settings.openAPIDocViewer &&
+        this.settings.openAPIDocViewer.openAPIDocType.swagger.enabled &&
+        this.settings.openAPIDocViewer.openAPIDocType.redoc.enabled;
+
+      if (this.page.type === 'SWAGGER' && !this.page.configuration.viewer) {
+        if (this.settings && this.settings.openAPIDocViewer) {
+          this.page.configuration.viewer = this.settings.openAPIDocViewer.openAPIDocType.defaultType;
+        }
+      }
+    };
+
+    this.openApiFormatLabel = (format) => {
+      if (this.settings && this.settings.openAPIDocViewer && format === this.settings.openAPIDocViewer.openAPIDocType.defaultType) {
+        return `${format} (Default)`;
+      } else {
+        return format;
+      }
+    };
+
+    this.toggleEntrypointAsServer = () => {
+      if (this.page.configuration.entrypointsAsServers === undefined) {
+        // Enable adding context-path automatically only the first time user decides to use entrypoint url.
+        this.page.configuration.entrypointAsBasePath = 'true';
+      }
+    };
+
+    this.usedAsGeneralConditions = () => {
+      return this.page.generalConditions;
+    };
+  },
+};
+
+export default EditPageAclsComponent;
```

Comment:
We should avoid default export and do `export const EditPageAclsComponent` direcly instead